src/structurizr/view/view_set.py

```diff
@@ -279,16 +291,50 @@ def create_dynamic_view(self, **kwargs) -> DynamicView:
         Args:
             **kwagrs: Provide keyword arguments for instantiating a `DynamicView`.
         """
-        # TODO:
-        # AssertThatTheViewKeyIsUnique(key);
         dynamic_view = DynamicView(**kwargs)
+        self._ensure_key_is_specific_and_unique(dynamic_view.key)
         dynamic_view.set_viewset(self)
         dynamic_view.set_model(self.model)
         self.dynamic_views.add(dynamic_view)
         return dynamic_view
 
+    def create_filtered_view(self, **kwargs) -> FilteredView:
+        """
+        Add a new FilteredView to the ViewSet.
+
+        Args:
+            **kwargs: Provide keyword arguments for instantiating a `FilteredView`.
+        """
+        filtered_view = FilteredView(**kwargs)
+        self._ensure_key_is_specific_and_unique(filtered_view.key)
+        filtered_view.set_viewset(self)
+        self.filtered_views.add(filtered_view)
+        return filtered_view
+
+    def get_view(self, key: str) -> Optional[AbstractView]:
+        """Return the view with the given key, or None."""
+        all_views = (
+            self.system_landscape_views
+            | self.system_context_views
+            | self.container_views
+            | self.component_views
+            | self.deployment_views
+            | self.dynamic_views
+            | self.filtered_views
+        )
```

Comment:
An alternative could be either:

```py
from itertools import chain

all_views = chain(self.system_landscape_views, ...)
```

or creating a `dict` to fetch the key from. I think the `chain` is nice because it avoids creation of an object that is then just iterated over and thrown away.

Reply:
Like it.